test/jdk/tools/jpackage/helpers/jdk/jpackage/test/JPackageCommand.java

```diff
@@ -742,6 +742,11 @@ public JPackageCommand validateOutput(Consumer<Iterator<String>> validator) {
         return this;
     }
 
+    public JPackageCommand setIgnoreExitCode(boolean v) {
+        ignoreExitCode = v;
+        return this;
+    }
```

Comment:
This is bad. It makes unclear semantics of the `JPackageCommand.execute(int expectedExitCode)` function.

I suggest adding `JPackageCommand.executeIgnoreExitCode()` function instead in the following way:
```
public Executor.Result executeIgnoreExitCode() {
    return execute(Optional.empty());
}

public Executor.Result execute(int expectedExitCode) {
    return execute(Optional.of(expectedExitCode));
} 

private Executor.Result execute(Optional expectedExitCode) {
    // Here goes the implementation of the current execute(int expectedExitCode) function adjusted accordingly
} 
```

Reply:
Fixed.